Detectors/GlobalTracking/include/GlobalTracking/TrackCuts.h

```diff
@@ -60,62 +86,68 @@ class TrackCuts
     auto src = trackIndex.getSource(); // make selections depending on track source
     // ITS tracks
     if (contributorsGID[GIndex::Source::ITS].isIndexSet()) { // ITS tracks selection
-      isBarrelTrack = true;
       const auto& itsTrk = data.getITSTrack(contributorsGID[GID::ITS]);
       int ITSnClusters = itsTrk.getNClusters();
       float ITSchi2 = itsTrk.getChi2();
       float itsChi2NCl = ITSnClusters != 0 ? ITSchi2 / (float)ITSnClusters : 0;
       uint8_t itsClusterMap = itsTrk.getPattern();
-      if (ITSnClusters <= mMinNClustersITS ||
+      if (itsTrk.getPt() <= mPtITSCut ||
+          std::abs(itsTrk.getEta()) > mEtaITSCut || // TODO: define 2 different values for min and max (**)
+          ITSnClusters <= mMinNClustersITS ||
           itsChi2NCl >= mMaxChi2PerClusterITS ||
           TrackMethods::FulfillsITSHitRequirements(itsClusterMap, mRequiredITSHits) == false) {
         return false;
       }
     }
     // TPC tracks
     if (contributorsGID[GIndex::Source::TPC].isIndexSet()) {
-      isBarrelTrack = true;
       const auto& tpcTrk = data.getTPCTrack(contributorsGID[GID::TPC]);
-      math_utils::Point3D<float> v{};
+      math_utils::Point3D<float> v{}; // vertex not defined?!
       std::array<float, 2> dca;
       if (tpcTrk.getPt() < mPtTPCCut ||
-          std::abs(tpcTrk.getEta()) > mEtaTPCCut ||
+          std::abs(tpcTrk.getEta()) > mEtaTPCCut || // TODO: define 2 different values for min and max (***)
           tpcTrk.getNClusters() < mNTPCClustersCut ||
-          (!(const_cast<o2::tpc::TrackTPC&>(tpcTrk).propagateParamToDCA(v, mBz, &dca, mDCACut)) ||
-           std::abs(dca[0]) > mDCACutY)) {
+          (!(const_cast<o2::tpc::TrackTPC&>(tpcTrk).propagateParamToDCA(v, mBz, &dca, mDCATPCCut)) ||
+           std::abs(dca[0]) > mDCATPCCutY) ||
+          std::hypot(dca[0], dca[1]) > mDCATPCCut) {
         return false;
       }
     }
-    if (isBarrelTrack) { // track selection for barrel tracks
+    // ITS-TPC matched cuts
+    // --> currently inactive in MatchITSTPCQC, since either GID::TPC or GID::ITS
+    if (src == o2::dataformats::GlobalTrackID::ITSTPC ||
+        src == o2::dataformats::GlobalTrackID::ITSTPCTRD ||
+        src == o2::dataformats::GlobalTrackID::ITSTPCTOF ||
+        src == o2::dataformats::GlobalTrackID::ITSTPCTRDTOF) { // track selection for barrel tracks (ITS-TPC matched)
       trk = data.getTrackParam(trackIndex);
-      if (trk.getPt() < mMinPt && trk.getPt() > mMaxPt && trk.getEta() < mMinEta && trk.getEta() > mMaxEta) {
+      if (trk.getPt() < mMinPt || trk.getPt() > mMaxPt || trk.getEta() < mMinEta || trk.getEta() > mMaxEta) {
```

Comment:
getEta is not a getter but does quite heavy calculation. If it is used more than once better assign it to local variable and reuse it.